### PR TITLE
Add user doc for etcd features

### PR DESCRIPTION
**Pull request by spzala**
Related https://github.com/etcd-io/etcd/pull/14483

Signed-off-by: Sahdev Zala <spzala@us.ibm.com>

### CODE REVIEW

**Comment by spzala** [comment]
cc @ahrtr @serathius

File: content/en/docs/v3.4/dev-guide/features.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Unsafe
+
+Unsafe features are rare and listed under the `Unsafe feature:` section in the etcd usage documentation. By default, they are disabled. They should be used with caution following documentation. An unsafe feature can be removed in the next minor or major release without following the feature deprecation policy.
+
+## Feature Deprecation
+
+### Experimental
+An experimental feature deprecates when it graduates to the stable stage.
+- The experimental feature documentation will show a deprecation message with a recommendation to use a related stable feature. e.g. `DEPRECATED. Use <feature-name> instead.`
+- A derprecated feature will be removed in the following release.
```
Comment (serathius) [inline content/en/docs/v3.4/dev-guide/features.md:37]:
```suggestion
- A deprecated feature will be removed in the following release.
```

**Review by serathius — APPROVED** [review]
Looks good overall.

File: content/en/docs/v3.4/dev-guide/features.md (hunk 1 above)
Comment (spzala) [reply content/en/docs/v3.4/dev-guide/features.md:37]: ah :) Good catch, thanks @serathius

File: content/en/docs/v3.4/dev-guide/features.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Experimental
+
+In order to get early feedback, any new feature is usually added as an experimental feature. The experimental feature can be identified by looking at the flag name, which should have `--experimental` as a prefix. Please consider the following points while using an experimental feature: 
+- It might be buggy due to a lack of user testing. Enabling the feature may not work as expected.
+- It is disabled by default. 
+- Support for such a feature may be dropped at any time without notice
+    - It can be removed in the next minor or major release without following the [feature deprecation](#feature-deprecation) policy unless it graduates to a stable future.
+    - The project team would appreciate users reporting any issues related to experimental features. However, such issues may be given lower priorities compared to the issues related to stable featuers.
+- An experimental [feature deprecates](#feature-deprecation) when it graduates to the stable stage. Users should start using a stable feature as soon as possible.
```
Comment (ahrtr) [inline content/en/docs/v3.4/dev-guide/features.md:19]:
```suggestion
- An experimental [feature flag deprecates](#feature-deprecation) when it graduates to the stable stage. Users should start using a stable feature flag as soon as possible.
```

File: content/en/docs/v3.4/dev-guide/features.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Unsafe
+
+Unsafe features are rare and listed under the `Unsafe feature:` section in the etcd usage documentation. By default, they are disabled. They should be used with caution following documentation. An unsafe feature can be removed in the next minor or major release without following the feature deprecation policy.
+
+## Feature Deprecation
+
+### Experimental
+An experimental feature deprecates when it graduates to the stable stage.
+- The experimental feature documentation will show a deprecation message with a recommendation to use a related stable feature. e.g. `DEPRECATED. Use <feature-name> instead.`
```
Comment (ahrtr) [inline content/en/docs/v3.4/dev-guide/features.md:36]:
```suggestion
- The experimental feature documentation will show a deprecation message with a recommendation to use a related stable feature flag. e.g. `DEPRECATED. Use <feature-name> instead.`
```

**Review by ahrtr — APPROVED** [review]
Overall looks good to me, with two minor comments. I'd defer to you whether to resolve them. 

Thanks @spzala .

File: content/en/docs/v3.4/dev-guide/features.md (hunk 2 above)
Comment (spzala) [reply content/en/docs/v3.4/dev-guide/features.md:19]: @ahrtr thanks for the quick review. This depends on how we read it but it sounds good to me this way too. I am updating accordingly.

Comment (ahrtr) [reply content/en/docs/v3.4/dev-guide/features.md]: ack.

As an non-native English speaker, previous sentence is a little confusing to me. It looked like the deprecated experimental feature and the graduated stable feature are two different features. But actually they are the same feature, they just have two different flags. 

It's much better now to me. 

Thank.

Comment (spzala) [reply content/en/docs/v3.4/dev-guide/features.md]: @ahrtr +1 thanks and sounds great!

**Pull request merged**